File: solutions/go/resistor-color/resistor_color.go
package resistorcolor

// Colors should return the list of all colors.
func Colors() []string {
	return []string{
		"black",
		"brown",
		"red",
		"orange",
		"yellow",
		"green",
		"blue",
		"violet",
		"grey",
		"white",
	}
}

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	for i, v := range Colors() {
		if color == v {
			return i
		}
	}
	return -1
}

File: solutions/python/resistor-color/resistor_color.py
_COLORS = ('black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white')


def color_code(color: str) -> int:
    return _COLORS.index(color)


def colors() -> list[str]:
    return list(_COLORS)
